### PR TITLE
Protected events settlement engine

### PR DESCRIPTION
**Pull request by ochaloup**
Update Claims pipe to support uptime/rug settlement based on PMPE

### CODE REVIEW

**Comment by ochaloup** [comment]
This is is ready for review to be later merged.

File: protected-event-distribution/src/utils.rs [hunk 1]
```diff
@@ -53,3 +54,17 @@ pub fn bps(value: u64, max: u64) -> u64 {
 pub fn bps_to_fraction(value: u64) -> f64 {
     value as f64 / 10000.0
 }
+
+/// pmpe is a gain in SOLs per 1000 of staked SOLs
+pub fn pmpe_to_epr(value: f64) -> f64 {
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply protected-event-distribution/src/utils.rs:59]: Fully correct. Buto already mentioned this to me and I was just ignorant here :sweat_smile:

File: protected-event-distribution/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+### 1. Gathering input data
+
+```bash
+# Download input files from Google Storage
+epoch=652
+bucket=marinade-validator-bonds-mainnet
+# generated by @marinade-finance/validator-bonds > snapshot-parser-cli
+gcloud storage cp "gs://$bucket/$epoch/stakes.json" "gs://$bucket/$epoch/validators.json" .
+gcloud storage cp "gs://$bucket/$epoch/stakes.json" "stakes.json"
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply protected-event-distribution/README.md:13]: Ah, f**k. Thanks.

File: .buildkite/prepare-protected-events.yml [hunk 3]
```diff
@@ -31,19 +33,81 @@ steps:
...
+  - label: ":cloud: :arrou_right: :floppy_disk: Downloading DS SAM auction data"
+    env:
+      # TODO: can we use authenticated requests for github api here? -H "Authorization: Bearer YOUR-TOKEN"
+      curl_command: 'curl -L -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28"'
+    commands:
+      - 'epoch=${EPOCH:-$(buildkite-agent meta-data get epoch)}'
+      # checking auction directory to work with
+      - |
+        epoch_auction_dirs=$(eval "$$curl_command $ds_sam_auction_github_api_link" |\
+          jq '.[] | .name' | tr -d '"' | grep "$$epoch")
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply .buildkite/prepare-protected-events.yml:56]: :sweat_smile:

**Comment by janlegner** [comment]
The logic of the PR looks good to me, just a few general-guidance

**Comment by ochaloup** [comment]
Thanks @janlegner. I updated the PR with the points you found. I would like to ask you for a re-review.
The most unclear thing is the naming of the `grace` var for commission.

**Review by janlegner — APPROVED** [review]

**Comment by ochaloup** [comment]
Squashing the changes and rebase fixes with fix for changes the package name (settlement engine -> protected-event-distirbution).

**Pull request merged**